packages/dart/test/parse_object_test.dart

```diff
@@ -0,0 +1,130 @@
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:parse_server_sdk/parse_server_sdk.dart';
+import 'package:test/test.dart';
+import 'parse_query_test.mocks.dart';
+
+@GenerateMocks([ParseClient])
+void main() {
+  group('parseObject', () {
+    test('should return expectedIncludeResult json when use fetch', () async {
+      // arrange
+      final MockParseClient client = MockParseClient();
+
+      await Parse().initialize(
+        'appId',
+        'https://test.parse.com',
+        debug: true,
+        // to prevent automatic detection
+        fileDirectory: 'someDirectory',
+        // to prevent automatic detection
+        appName: 'appName',
+        // to prevent automatic detection
+        appPackageName: 'somePackageName',
+        // to prevent automatic detection
+        appVersion: 'someAppVersion',
+      );
+
+      ParseObject myUserObject = ParseObject("MyUser", client: client);
+      myUserObject.objectId = "Mn1iJTkWTE";
+
+      when(client.get(
+        any,
+        options: anyNamed("options"),
+        onReceiveProgress: anyNamed("onReceiveProgress"),
+      )).thenAnswer((_) async => ParseNetworkResponse(
+          statusCode: 200,
+          data:
+              "{\"results\":[{\"objectId\":\"Mn1iJTkWTE\",\"phone\":\"+12025550463\",\"createdAt\":\"2022-09-04T13:35:20.883Z\",\"updatedAt\":\"2022-11-14T10:55:56.202Z\",\"img\":{\"objectId\":\"8nGrLj3Mvk\",\"size\":67663,\"mime\":\"image/jpg\",\"file\":{\"__type\":\"File\",\"name\":\"dc7320ee9146ee19aed8997722fd4e3c.bin\",\"url\":\"http://ip:port/api/files/myapp/dc7320ee9146ee19aed8997722fd4e3c.bin\"},\"createdAt\":\"2022-11-14T10:55:56.025Z\",\"updatedAt\":\"2022-11-14T10:55:56.025Z\",\"__type\":\"Object\",\"className\":\"MyFile\"}}]}"));
```

Comment:
It is true that it can be written better

Reply:
@mtrezza 
This json is a fake json from a fake server

Reply:
Yes, but using raw JSON strings could make test debugging more difficult in the future or the test less robust. Could you create a normal object and stringily it to JSON?

Reply:
No, because JSON is sent from the server and should be used to simulate server data. If we want to create an object, the test model must be completely changed.

Reply:
I don't mean a Parse object, I mean a plain object and for example:

```dart
import 'dart:convert';
var response = {
     "key":"value"
};
String fakeServerResponseJson = jsonEncode(fakeServerResponse);
```

Reply:
Thanks; the idea is that it's easier to read, modify and debug in the future. We usually never use raw JSON strings in tests but convert them from an object.

Reply:
This method works, I will try to improve all the tests

Reply:
All tests improved

Reply:
Looks good!